### PR TITLE
Allow pgstac parameters for geojsonsearch to be globally set in Settings

### LINKED ISSUE #185: add Settings to set `pgstac.geojsonsearch` defaults
https://github.com/stac-utils/titiler-pgstac/blob/d353b9df7b8b2ff164258d94c014386a34d3ec89/titiler/pgstac/mosaic.py#L276-L280

ref https://github.com/stac-utils/titiler-pgstac/discussions/184#discussioncomment-10465575

### PR DESCRIPTION
**Pull request by zstatmanweil**
# Description
This closes #185. I have created a Settings object so the parameters to PgStac's `geojsonsearch` can be set globally via environment variables. I added an example to `.env.example` as well so hopefully it is clear how to set them. These can still also be set via query params. 

# Tests
I ran locally and made sure any change to these pgstac env variables would apply to any request I made. I then made sure I could pass in query params to a specific request and just that request would use that change. I confirmed those both by adding print statements in the code [here](https://github.com/stac-utils/titiler-pgstac/blob/main/titiler/pgstac/mosaic.py#L281), and working with my data that had issues when `existwhenfull` and `skipcovered` were both True. So easy for me to visually ensure what I expected to happen would happen.

### CODE REVIEW

**Comment by zstatmanweil** [comment]
@vincentsarago, question for you: when I open a PR in one of your projects, do I make updates to the CHANGES file at that time? Or do you update that when releasing a new version?

**Comment by zstatmanweil** [comment]
Also, I tried to follow instructions for running tests but I get this error for all of them: "psycopg.errors.FeatureNotSupported: extension "postgis" is not available"

**Comment by vincentsarago** [comment]
🙏 @zstatmanweil 

> @vincentsarago, question for you: when I open a PR in one of your projects, do I make updates to the CHANGES file at that time? Or do you update that when releasing a new version?

It depends, I always appreciate a sentence in a changelog within an `Unreleased` section 

> Also, I tried to follow instructions for running tests but I get this error for all of them: "psycopg.errors.FeatureNotSupported: extension "postgis" is not available"

Interesting, I'll have a look

File: titiler/pgstac/settings.py [hunk 1]
```diff
@@ -111,6 +111,31 @@ def check_enable(self):
...
+    time_limit: int = 5
+
+    # Return as soon as the geometry is fully covered
+    exitwhenfull: bool = True
+
+    # Skip any items that would show up completely under the previous items
+    skipcovered: bool = True
+
+    model_config = {
+        "env_prefix": "TITILER_PGSTAC_BACKEND_",
```
Comment (vincentsarago) [inline titiler/pgstac/settings.py:133]: @zstatmanweil what do you think about `TITILER_PGSTAC_SEARCH_` prefix instead of `_BACKEND_`

Comment (zstatmanweil) [reply titiler/pgstac/settings.py]: I think that is better - updated!

**Comment by zstatmanweil** [comment]
I also added a line under `Unreleased` in the changes file

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**